src/test/java/core/basesyntax/bookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,278 @@
+package core.basesyntax.bookstore.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.nullable;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import core.basesyntax.bookstore.dto.book.BookDto;
+import core.basesyntax.bookstore.dto.book.BookParamsDto;
+import core.basesyntax.bookstore.dto.book.BookWithoutCategoryDto;
+import core.basesyntax.bookstore.dto.book.CreateBookRequestDto;
+import core.basesyntax.bookstore.dto.book.UpdateBookRequestDto;
+import core.basesyntax.bookstore.exception.EntityNotFoundException;
+import core.basesyntax.bookstore.mapper.BookMapper;
+import core.basesyntax.bookstore.model.Book;
+import core.basesyntax.bookstore.repository.SpecificationBuilder;
+import core.basesyntax.bookstore.repository.book.BookRepository;
+import core.basesyntax.bookstore.repository.category.CategoryRepository;
+import core.basesyntax.bookstore.service.impl.BookServiceImpl;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceTest {
```

Comment:
the same comments. Let’s increase test readability. Move the objects to private methods or Util class and follow Given-When-Then ordering